File: product_frontend/src/services/ProductService.js
import axios from "axios";

export const getListProducts = async (page, search, productTypeId) => {
    try {
        let result;
        if (productTypeId) {
            result = await axios.get(`http://localhost:8080/api/product?page=${page}&searchName=${search}&productTypeId=${productTypeId}`);
            console.log(result);
        } else {
            result = await axios.get(`http://localhost:8080/api/product?page=${page}&searchName=${search}`);
            console.log(2);
            console.log(result);


        }
      return result;
    } catch (e) {
        console.log(e);
    }
}
export const editProduct = async (product) => {
    try {
        const result = await axios.put(`http://localhost:8080/api/edit`, product);
        return result.data;
    } catch (e) {
        console.log(e);
    }
}
export const getById = async (id) => {
    try {
        const result = await axios.get(`http://localhost:8080/api/${id}`);
        console.log(result.data)
        return result.data;
    } catch (e) {
        console.log(e);
    }
}
export const deleteProduct = async (id) => {
    try {
        const result = await axios.delete(`http://localhost:8080/api/delete/${id}`);
        return result.data
    } catch (e) {
        console.log(e);
    }
}
export const addProduct = async (product) => {
    try {
        const result = await axios.post(`http://localhost:8080/api/create`, product);
        return result.data;
    } catch (e) {
        console.log(e);
    }
}

File: product_backend/product/src/main/java/com/example/product/service/product/ProductService.java
package com.example.product.service.product;

import com.example.product.modal.Product;
import com.example.product.repository.ProductRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

@Service
public class ProductService implements IProductService {
    @Autowired
    private ProductRepository productRepository;
    @Override
    public Page<Product> getListProduct(Pageable pageable, String search) {

        return productRepository.findAllProduct(pageable,"%"+search+"%");
    }

    @Override
    public void add(Product product) {

    }

    @Override
    public void delete(Integer id) {
         productRepository.deleteProduct(id);
    }

    @Override
    public void edit(Product product) {
        productRepository.updateProduct(product.getName(), product.getStartDay(), product.getQuantity(), product.getId(),product.getProductType().getId());
    }

    @Override
    public Product getById(Integer id) {
        return productRepository.getById(id);
    }
}
